### DISCUSSION TITLE
use  WorldCRS84Quad  reload map

### DISCUSSION THREAD

**Discussion by liuziqian1** [developmentseed/titiler#940] [Q&A]
reload  http://127.0.0.1:8000/cog/tiles/WorldCRS84Quad/{z}/{x}/{y}.png?url=https://s3.amazonaws.com/opendata.remotepixel.ca/cogs/natural_earth/world.tif

**Comment by liuziqian1** [comment]
http://127.0.0.1:8000/cog/tiles/WorldCRS84Quad/{z}/{x}/{y}.png?url=https://s3.amazonaws.com/opendata.remotepixel.ca/cogs/natural_earth/world.tif   加载这个地址，只展示了北美的区域，滑不到其他地区

**Comment by liuziqian1** [comment]
![screenshot-20240626-200437](https://github.com/developmentseed/titiler/assets/44964612/0b09d848-e89c-4931-912d-4298cb155c8d)

**Comment by vincentsarago** [comment]
Dear @liuziqian1 please state what's the issue.

**Reply by liuziqian1** [reply]
I loaded the TIFF file provided in the document using this address, but I found that only North America is shown, and it's distorted.

It should normally be like this.
![Uploading screenshot-20240626-203243.png…]()

**Reply by vincentsarago** [reply]
Because you're trying to use the `WorldCRS84Quad` TileMatrixSet with Mapbox GL which only support the WebMercatorQuad TMS. 

if you want to use this TMS, you'll need to use either leaflet (https://titiler.xyz/cog/WorldCRS84Quad/map?url=https://s3.amazonaws.com/opendata.remotepixel.ca/cogs/natural_earth/world.tif) or open layer

**Reply by liuziqian1** [reply]
But why is it that when I use QGIS to load and display directly, it is not complete?

**Reply by liuziqian1** [reply]
![screenshot-20240627-091700](https://github.com/developmentseed/titiler/assets/44964612/28cb3aa9-3f97-4327-b140-f36776ad7bb6)

**Reply by liuziqian1** [reply]
![screenshot111-20240627-093027](https://github.com/developmentseed/titiler/assets/44964612/7ef75290-720d-497a-be56-d3aa458fb858)

**Reply by liuziqian1** [reply]
Do you have openlayer demo ?

**Reply by vincentsarago** [reply]
> But why is it that when I use QGIS to load and display directly, it is not complete?

you think it displays it correctly but it doesn't! QGIS still thinks the XYZ layer is using the WebMercatorQuad TileMatrixSet but doesn't wrap at the end of the world like Mapbox or Leaflet can do. 

> Do you have openlayer demo ?

No

**Reply by liuziqian1** [reply]
https://titiler.xyz/cog/info?url=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/86bea8fa8d2de833480f9f903c82211b/%E9%81%A5%E6%84%9F%E5%BD%B1%E5%83%8F/baoding_2024_0421-0000000000-0000000000_cog.tif    
The loading of this cog address results in a completely black screen.

**Reply by liuziqian1** [reply]
I load map   use this  :    http://127.0.0.1:8000/cog/tiles/WebMercatorQuad/{z}/{x}/{y}.png?url=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/52086c726d2b2bb6f2c09040cc784d31/秦皇岛/qinhuiangdao_4_image_cog.tif&bidx=4&bidx=1&bidx=2

**Reply by vincentsarago** [reply]
your COG is of type UINT16 
```
curl https://titiler.xyz/cog/info\?url\=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/86bea8fa8d2de833480f9f903c82211b/%E9%81%A5%E6%84%9F%E5%BD%B1%E5%83%8F/baoding_2024_0421-0000000000-0000000000_cog.tif | jq -c '.dtype' 

"uint16"
```

So you'll need to `rescale` the cog values between a Min/Max in order to see something 

```
curl https://titiler.xyz/cog/statistics\?url\=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/86bea8fa8d2de833480f9f903c82211b/%E9%81%A5%E6%84%9F%E5%BD%B1%E5%83%8F/baoding_2024_0421-0000000000-0000000000_cog.tif  | jq '.b1'    

{
  "min": 0.0,
  "max": 18191.0,
  "mean": 968.1481425572235,
  "count": 861184.0,
  "sum": 833753690.0,
  "std": 904.3577313102603,
  "median": 1391.0,
  "majority": 0.0,
  "minority": 58.0,
  "unique": 4239.0,
  "histogram": [
    [
      715274.0,
      144921.0,
      454.0,
      201.0,
      132.0,
      110.0,
      63.0,
      20.0,
      5.0,
      4.0
    ],
    [
      0.0,
      1819.1,
      3638.2,
      5457.299999999999,
      7276.4,
      9095.5,
      10914.599999999999,
      12733.699999999999,
      14552.8,
      16371.9,
      18191.0
    ]
  ],
  "valid_percent": 100.0,
  "masked_pixels": 0.0,
  "valid_pixels": 861184.0,
  "percentile_2": 0.0,
  "percentile_98": 2376.0
}
```

☝️ show some statistics of your file, you can see that the 98th percentile of band 1 is around 2300 DN (almost the same for the other bands)

so let's use `0 -> 3000` for min/max rescaling 

`https://titiler.xyz/cog/statistics\?url\=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/86bea8fa8d2de833480f9f903c82211b/%E9%81%A5%E6%84%9F%E5%BD%B1%E5%83%8F/baoding_2024_0421-0000000000-0000000000_cog.tif&rescale=0,3000`

<img width="1351" src="https://github.com/developmentseed/titiler/assets/10407788/08c5f578-0989-48d2-a00b-9c8fb4bd772b">

**Reply by liuziqian1** [reply]
http://127.0.0.1:8000/cog/tiles/WebMercatorQuad/8/213/97.png?url=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/52086c726d2b2bb6f2c09040cc784d31/%E7%A7%A6%E7%9A%87%E5%B2%9B/qinhuiangdao_4_image_cog.tif&bidx=3&bidx=2&bidx=1&rescale=0,3000

It is error  500

**Reply by vincentsarago** [reply]
Also noticing that you have 4 bands in your dataset, so you'll need to select 1 or 3 valid bands in order to create a proper visual representation of your dataset 

`https://titiler.xyz/cog/WorldCRS84Quad/map?url=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/86bea8fa8d2de833480f9f903c82211b/遥感影像/baoding_2024_0421-0000000000-0000000000_cog.tif&rescale=0,7000&bidx=3&bidx=2&bidx=1`

<img width="1230" src="https://github.com/developmentseed/titiler/assets/10407788/930f609a-eb18-4e5e-afd8-c2fef0889349">


or calculating NDVI ;-) 

`https://titiler.xyz/cog/WorldCRS84Quad/map?url=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/86bea8fa8d2de833480f9f903c82211b/遥感影像/baoding_2024_0421-0000000000-0000000000_cog.tif&rescale=-0.5,1&expression=%28%28b4-b3%29%2F%28b3%2Bb4%29%29&colormap_name=viridis`

<img width="638" src="https://github.com/developmentseed/titiler/assets/10407788/72a15a34-837b-4ef5-9494-d1adad13a916">

**Reply by vincentsarago** [reply]
🤷 works here https://titiler.xyz/cog/tiles/WebMercatorQuad/8/213/97.png?url=https://xtgc.geo-compass.com/uploadedFiles/mergeFile/52086c726d2b2bb6f2c09040cc784d31/秦皇岛/qinhuiangdao_4_image_cog.tif&bidx=3&bidx=2&bidx=1&rescale=0,3000

**Comment by liuziqian1** [comment]
If I need to set the color to red when band 1 is greater than 5 and band 2 is less than 5, and display transparent in all other cases, how should I configure this?

**Reply by vincentsarago** [reply]
You need to use an expression to first create a value from those bands  `expression=(b1>5)&(b2<5)` and then apply a custom colormap https://developmentseed.org/titiler/advanced/rendering/#color-map

**Comment by liuziqian1** [comment]
{
  "b1": {
    "min": 0.0116130672395229,
    "max": 0.779452502727509,
    "mean": 0.0601098776788135,
    "count": 324034,
    "sum": 19477.6441037767,
    "std": 0.0344225130845347,
    "median": 0.0491757690906525,
    "majority": 0.0282334592193365,
    "minority": 0.0116130672395229,
    "unique": 320964,
    "histogram": [
      [272905, 47157, 3396, 353, 85, 62, 52, 17, 2, 5],
      [0.0116130672395229, 0.0883970111608505, 0.165180951356888, 0.241964891552925, 0.318748831748962, 0.395532786846161, 0.472316741943359, 0.549100697040558, 0.625884592533112, 0.70266854763031, 0.779452502727509]
    ],
    "valid_percent": 34.77,
    "masked_pixels": 607806,
    "valid_pixels": 324034,
    "percentile_2": 0.0254898574203253,
    "percentile_98": 0.153214901685715
  },
  "b2": {
    "min": 0.024211049079895,
    "max": 0.822443604469299,
    "mean": 0.0823860980792293,
    "count": 324034,
    "sum": 26695.896905005,
    "std": 0.0334502484495934,
    "median": 0.0722084641456604,
    "majority": 0.0528662949800491,
    "minority": 0.024211049079895,
    "unique": 319302,
    "histogram": [
      [264531, 56148, 2882, 248, 78, 55, 62, 21, 5, 4],
      [0.024211049079895, 0.104034304618835, 0.183857560157776, 0.263680815696716, 0.343504071235657, 0.423327326774597, 0.503150582313538, 0.582973837852478, 0.662797093391419, 0.742620348930359, 0.822443604469299]
    ],
    "valid_percent": 34.77,
    "masked_pixels": 607806,
    "valid_pixels": 324034,
    "percentile_2": 0.0473339594900608,
    "percentile_98": 0.170403957366943
  },
  "b3": {
    "min": 0.0136609524488449,
    "max": 0.899312734603882,
    "mean": 0.0735884339807258,
    "count": 324034,
    "sum": 23845.1546165105,
    "std": 0.0451956096141563,
    "median": 0.0575872138142586,
    "majority": 0.0345752090215683,
    "minority": 0.0136609524488449,
    "unique": 321374,
    "histogram": [
      [261490, 55178, 6795, 347, 89, 65, 53, 7, 7, 3],
      [0.0136609524488449, 0.102226130664349, 0.190791308879852, 0.279356479644775, 0.367921650409699, 0.456486850976944, 0.54505205154419, 0.633617222309113, 0.722182393074036, 0.810747563838959, 0.899312734603882]
    ],
    "valid_percent": 34.77,
    "masked_pixels": 607806,
    "valid_pixels": 324034,
    "percentile_2": 0.0289180371910334,
    "percentile_98": 0.193758830428123
  },
  "b4": {
    "min": 0.0166560485959053,
    "max": 0.911129236221314,
    "mean": 0.350964477644063,
    "count": 324034,
    "sum": 113724.423548916,
    "std": 0.0582608785108788,
    "median": 0.364042907953262,
    "majority": 0.370503544807434,
    "minority": 0.0166560485959053,
    "unique": 315374,
    "histogram": [
      [100, 3851, 44636, 140917, 131809, 2274, 412, 26, 6, 3],
      [0.0166560485959053, 0.106103368103504, 0.195550680160522, 0.284997999668121, 0.37444531917572, 0.463892638683319, 0.553339958190918, 0.642787277698517, 0.732234597206116, 0.821681916713715, 0.911129236221314]
    ],
    "valid_percent": 34.77,
    "masked_pixels": 607806,
    "valid_pixels": 324034,
    "percentile_2": 0.212172403931618,
    "percentile_98": 0.438589692115784
  }
}

**Reply by liuziqian1** [reply]
The information returned by using /cog/statistics on my TIFF file is as follows。 my COG is of type float32  . How to load it

**Reply by liuziqian1** [reply]
I now load it use http://172.16.100.1:81/cog/tiles/WebMercatorQuad/14/13435/6320.png?url=http://10.13.30.93:8080/uploadedFiles/mergeFile/8cadaf1f21bd7066a47604416dd12bae/xinji/xinjishi2024081_cog.tif&rescale=0,7000&bidx=3&bidx=2&bidx=1&nodata=0    ,   but  it is black all
